streams/src/test/java/org/apache/kafka/streams/integration/KStreamRepartitionIntegrationTest.java

```diff
@@ -0,0 +1,859 @@
+        assertEquals(4, getNumberOfPartitionsForTopic(repartitionTopicName));
+    }
+
+    @Test
+    public void shouldCreateOnlyOneRepartitionTopicWhenRepartitionIsFollowedByGroupByKey() throws ExecutionException, InterruptedException {
+        final String repartitionName = "new-partitions";
+        final long timestamp = System.currentTimeMillis();
+
+        sendEvents(
+            timestamp,
+            Arrays.asList(
+                new KeyValue<>(1, "A"),
+                new KeyValue<>(2, "B")
+            )
+        );
+
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final Repartitioned<String, String> repartitioned = Repartitioned.<String, String>as(repartitionName)
+            .withKeySerde(Serdes.String())
+            .withValueSerde(Serdes.String())
+            .withNumberOfPartitions(1);
+
+        builder.stream(inputTopic, Consumed.with(Serdes.Integer(), Serdes.String()))
+               .selectKey((key, value) -> key.toString())
+               .repartition(repartitioned)
+               .groupByKey()
+               .count()
+               .toStream()
+               .to(outputTopic);
+
+        startStreams(builder);
+
+        final String topology = builder.build().describe().toString();
+
+        validateReceivedMessages(
+            new StringDeserializer(),
+            new LongDeserializer(),
+            Arrays.asList(
+                new KeyValue<>("1", 1L),
+                new KeyValue<>("2", 1L)
+            )
+        );
+
+        assertTrue(topicExists(toRepartitionTopicName(repartitionName)));
+        assertEquals(1, countOccurrencesInTopology(topology, "Sink: .*-repartition"));
+    }
+
+    @Test
+    public void shouldGenerateRepartitionTopicWhenNameIsNotSpecified() throws ExecutionException, InterruptedException {
+        final long timestamp = System.currentTimeMillis();
+
+        sendEvents(
+            timestamp,
+            Arrays.asList(
+                new KeyValue<>(1, "A"),
+                new KeyValue<>(2, "B")
+            )
+        );
+
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        builder.stream(inputTopic, Consumed.with(Serdes.Integer(), Serdes.String()))
+               .selectKey((key, value) -> key.toString())
+               .repartition(Repartitioned.with(Serdes.String(), Serdes.String()))
+               .to(outputTopic);
+
+        startStreams(builder);
+
+        validateReceivedMessages(
+            new StringDeserializer(),
+            new StringDeserializer(),
+            Arrays.asList(
+                new KeyValue<>("1", "A"),
+                new KeyValue<>("2", "B")
+            )
+        );
+
+        final String topology = builder.build().describe().toString();
+
+        assertEquals(1, countOccurrencesInTopology(topology, "Sink: .*-repartition"));
+    }
+
+    @Test
+    public void shouldGoThroughRebalancingCorrectly() throws ExecutionException, InterruptedException {
```

Comment:
Not sure what this test is about, ie, how does is relate to the `repartition()` feature?

Reply:
It's related to this comment https://github.com/apache/kafka/pull/7170#issuecomment-522163447

Reply:
Thanks for clarifying!